S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -52,40 +54,72 @@ public class GameNote : S2VXNote {
             Delete();
         }
 
-        // Notes are clickable if they are visible on screen and not missed
+        // Notes are clickable if they are visible on screen, not missed, and is the earliest note
         private bool IsClickable() {
             var time = Time.Current;
-            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run) 
+            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run)
             TimingError = (int)Math.Round(Math.Clamp(time - EndTime, -MissThreshold, MissThreshold));
-            return TimingError <= MissThreshold && Alpha > 0;
+            var inMissThreshold = TimingError <= MissThreshold && Alpha > 0;
+            var earliestNote = (GameNote)Story.Notes.Children[^1];
```

Comment:
Don't think you need the GameNote cast, leaving it as the default S2VXNote should be fine. I believe an == check by default for reference?

Reply:
Oh ok